### PR TITLE
[draft] Feature/wishlist

### PR DESCRIPTION
**Pull request by timhofman**

### CODE REVIEW

File: packages/magento-wishlist/components/ProductWishlistChip/index.tsx [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+
+  const { data: token } = useQuery(CustomerTokenDocument)
+  const isLoggedIn = token?.customerToken && token?.customerToken.valid
+
+  const heart = (
+    <IconSvg
+      src={iconHeart}
+      size='medium'
+      className={classes.iconHeart}
+      sx={(theme) => ({ stroke: '#AC2E2E' })}
```
Comment (ErwinOtten) [inline packages/magento-wishlist/components/ProductWishlistChip/index.tsx:43]: - Use color attribute to define color, not stroke. (IconSvg has `stroke: currentColor`)
- Use a color from the theme.palette object
- Isn't theme.palette.text.primary the better solution here? Or: theme.palette.text.secondary

File: packages/magento-wishlist/components/ProductWishlistChip/index.tsx [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    />
+  )
+
+  const activeHeart = (
+    <IconSvg
+      src={iconHeart}
+      size='medium'
+      className={classes.iconHeartActive}
+      sx={{ stroke: '#AC2E2E', fill: '#AC2E2E' }}
```
Comment (ErwinOtten) [inline packages/magento-wishlist/components/ProductWishlistChip/index.tsx:52]: - Use color attribute to define color, not stroke
- add `sx={{fill: currentColor}}`
- Isn't color: theme.palette.primary.main the better solution here, or: theme.palette.text.primary

File: examples/magento-graphcms/components/Layout/LayoutFull.tsx [hunk 3]
```diff
@@ -89,6 +95,13 @@ export function LayoutFull(props: LayoutFullProps) {
             >
               <Trans>Customer Service</Trans>
             </MenuFabSecondaryItem>,
+            <MenuFabSecondaryItem
+              key='wishlist'
+              icon={<IconSvg src={iconHeart} size='medium' />}
+              href='/wishlist'
```
Comment (ErwinOtten) [inline examples/magento-graphcms/components/Layout/LayoutFull.tsx:101]: Can we add a Dot badge https://mui.com/components/badges/#dot-badge when Wishlist has items

File: examples/magento-graphcms/components/Layout/LayoutFull.tsx [hunk 4]
```diff
@@ -63,6 +64,11 @@ export function LayoutFull(props: LayoutFullProps) {
                 <IconSvg src={iconCustomerService} size='large' />
               </Fab>
             </PageLink>
+            <PageLink href='/wishlist' passHref>
+              <Fab aria-label={t`Wishlist`} size='large' color='inherit'>
+                <IconSvg src={iconHeart} size='large' />
+              </Fab>
+            </PageLink>
```
Comment (paales) [inline examples/magento-graphcms/components/Layout/LayoutFull.tsx:71]: Implement a badge: If you take a look at the <CustomerFab /> you'll see an implementation how it has been handled here to add a DesktopHeaderBadge (this one is style differently from a regular badge)

File: packages/magento-product/components/ProductAddToCart/ProductAddToCart.tsx [hunk 5]
```diff
@@ -16,12 +16,14 @@ import { AnimatePresence } from 'framer-motion'
...
 import React from 'react'
 import { ProductAddToCartDocument, ProductAddToCartMutationVariables } from './ProductAddToCart.gql'
+import { ProductWishlistChip } from '@graphcommerce/magento-wishlist'
 
 const { classes, selectors } = extendableComponent('ProductAddToCart', [
   'root',
   'button',
   'price',
   'divider',
+  'buttonWrapper',
```
Comment (paales) [inline packages/magento-product/components/ProductAddToCart/ProductAddToCart.tsx:26]: 👌

File: packages/magento-wishlist/components/ProductWishlistChip/index.tsx [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+
+export type ProductWishlistChipProps = ProductWishlistChipFragment &
+  IconButtonProps &
+  ProductWishlistSettings & { sx?: SxProps<Theme> }
+
+const name = 'ProductWishlistChip' as const
+const parts = ['root', 'iconHeart', 'iconHeartActive', 'wishlistButton'] as const
+const { classes } = extendableComponent(name, parts)
+
+export default function ProductWishlistChip(props: ProductWishlistChipProps) {
```
Comment (paales) [inline packages/magento-wishlist/components/ProductWishlistChip/index.tsx:29]:
```suggestion
export function ProductWishlistChip(props: ProductWishlistChipProps) {
```

File: packages/magento-wishlist/components/WishlistItem/index.tsx [hunk 7]
```diff
@@ -0,0 +1,281 @@
...
+  'productLink',
+  'image',
+  'itemName',
+  'itemPrice',
+  'discountPrice',
+  'root',
+] as const
+const { classes } = extendableComponent<OwnerState, typeof compName, typeof parts>(compName, parts)
+
+export default function WishlistItem(props: WishlistItemProps) {
```
Comment (paales) [inline packages/magento-wishlist/components/WishlistItem/index.tsx:44]:
```suggestion
export function WishlistItem(props: WishlistItemProps) {
```

File: packages/magento-wishlist/components/WishlistItems/index.tsx [hunk 8]
```diff
@@ -0,0 +1,25 @@
+import { AnimatedRow, RenderType, TypeRenderer } from '@graphcommerce/next-ui'
+import { AnimatePresence } from 'framer-motion'
+import React from 'react'
+import { WishlistItem } from '..'
+import { WishlistItemFragment } from '../WishlistItem/ProductWishlistItem.gql'
+
+export type WishlistProps = { items?: [WishlistItemFragment] }
+
+export default function WishlistItems(props: WishlistProps) {
```
Comment (paales) [inline packages/magento-wishlist/components/WishlistItems/index.tsx:9]:
```suggestion
export function WishlistItems(props: WishlistProps) {
```

File: packages/magento-wishlist/components/index.ts [hunk 9]
```diff
@@ -0,0 +1,3 @@
+export { default as ProductWishlistChip } from './ProductWishlistChip'
+export { default as WishlistItems } from './WishlistItems'
+export { default as WishlistItem } from './WishlistItem'
```
Comment (paales) [inline packages/magento-wishlist/components/index.ts:3]:
```suggestion
export * from './ProductWishlistChip'
export * from './WishlistItems'
export * from './WishlistItem'
```

File: packages/magento-wishlist/hooks/useMergeGuestWishlistWithCustomer.tsx [hunk 10]
```diff
@@ -0,0 +1,31 @@
...
+
+    if (!wishlist.length) return
+
+    let payload = wishlist.map((item) => ({
+      sku: item,
+      quantity: 1,
+    }))
+
+    localStorage.setItem(GUEST_WISHLIST_STORAGE_NAME, JSON.stringify([]))
+    addWishlistItem({ variables: { input: payload } })
```
Comment (paales) [inline packages/magento-wishlist/hooks/useMergeGuestWishlistWithCustomer.tsx:29]: What happens when an SKU doesn't exist anymore, does this throw? Maybe fetch the products first and _then_ add them to the wishlist?

File: yarn.lock [hunk 11]
```diff
@@ -1606,25 +1606,6 @@
     minimatch "^3.0.4"
```
Comment (paales) [inline yarn.lock:1606]: I wouldn't expect any changes in the yarn.lock. Replace the yarn.lock, stage changes, run `yarn` again -> should have no changes.

File: packages/magento-wishlist/components/ProductWishlistChip/index.tsx [hunk 12]
```diff
@@ -0,0 +1,139 @@
...
+import { CustomerTokenDocument } from '@graphcommerce/magento-customer'
+import {
+  GUEST_WISHLIST_STORAGE_NAME,
+  GetIsInWishlistsDocument,
+} from '@graphcommerce/magento-wishlist'
+import { ProductWishlistChipFragment } from './ProductWishlistChip.gql'
+import { t } from '@lingui/macro'
+
+type ProductWishlistSettings = {
+  display?: 'guest' | 'customer'
```
Comment (paales) [inline packages/magento-wishlist/components/ProductWishlistChip/index.tsx:18]: This isn't used anywhere at the moment? No problem though

File: examples/magento-graphcms/pages/wishlist/index.tsx [hunk 13]
```diff
@@ -0,0 +1,142 @@
...
+  )
+
+  if (isLoggedIn) {
+    wishlistItems =
+      GetCustomerWishlistData?.customer?.wishlists[0]?.items_v2?.items.map(
+        (item) => item?.product,
+      ) || []
+  } else {
+    wishlistItems = productGuestItems?.products?.items
+  }
```
Comment (paales) [inline examples/magento-graphcms/pages/wishlist/index.tsx:74]: This should all be moved to the library, the examples shouldn't contain any business logic.

File: examples/magento-graphcms/pages/wishlist/index.tsx [hunk 14]
```diff
@@ -0,0 +1,142 @@
...
+              }
+            >
+              <Trans>Discover our collection and add items to your wishlist!</Trans>
+            </FullPageMessage>
+          ) : (
+            <>
+              <FullPageMessage
+                title={t`Wishlist`}
+                icon={<IconSvg src={iconHeart} size='xl' />}
+              ></FullPageMessage>
```
Comment (paales) [inline examples/magento-graphcms/pages/wishlist/index.tsx:110]: Use `<LayoutTitle icon={bla} />`

**Review by paales — CHANGES_REQUESTED** [review]

**Comment by paales** [comment]
Maybe also incorporate

```
fragment WishlistStoreConfig on StoreConfig @inject(into: ["StoreConfigFragment"]) {
    magento_wishlist_general_is_enabled
}
```

**Review by paales — CHANGES_REQUESTED** [review]

**Pull request merged**